tests/BaseBuilderTest.php

```diff
@@ -5,29 +5,61 @@
 namespace Doctrine\Tests\RST;
 
 use Doctrine\RST\Builder;
+use Doctrine\RST\Configuration;
+use Doctrine\RST\ErrorManager;
+use Doctrine\RST\ErrorManager\ErrorManagerFactory;
 use Exception;
+use PHPUnit\Framework\MockObject\MockObject;
 use PHPUnit\Framework\TestCase;
 
 use function file_get_contents;
 use function shell_exec;
 
 abstract class BaseBuilderTest extends TestCase
 {
-    /** @var Builder */
-    protected $builder;
+    protected Configuration $configuration;
+    protected Builder $builder;
+    /** @var ErrorManager|MockObject */
+    protected $errorManager;
 
     abstract protected function getFixturesDirectory(): string;
 
     protected function setUp(): void
     {
         shell_exec('rm -rf ' . $this->targetFile());
 
-        $this->builder = new Builder();
-        $this->builder->getConfiguration()->setUseCachedMetas(false);
+        $this->configuration = new Configuration();
+        $this->configuration->setUseCachedMetas(false);
+        $this->errorManager  = $this->createMock(ErrorManager::class);
+        $errorManagerFactory = $this->createMock(ErrorManagerFactory::class);
+        $this->configuration->setErrorManagerFactory($errorManagerFactory);
+        $errorManagerFactory->method('getErrorManager')->willReturn($this->errorManager);
+        $this->configureExpectedWarnings();
+        $this->configureExpectedErrors();
+        $this->builder = new Builder($this->configuration);
         $this->configureBuilder($this->builder);
+        //build mus be executed
         $this->builder->build($this->sourceFile(), $this->targetFile());
     }
 
+    /**
+     * We expect no warnings to occur in building. If you want to test
```

Comment:
"in building" reads like inside the physical place where you work. I'd use either "during the build" or "while building the documentation"